### DISCUSSION TITLE
How to make a simple inline Bot using tgbot-cpp?

### DISCUSSION THREAD

**Discussion by Amirhan-Taipovjan-Greatest-I** [reo7sp/tgbot-cpp#318] [Q&A] [answered]
Hello!

I want to make inline Feature for My own Bot, but I actually can't figure out how to do this...
My Goal is to make a inline Feature where User can write down some text and get another text as output after Confirming.

I've tried to search in there or somewhere else in the Internet... anything that can help Me, but for C++ I didn't find it...

Thanks for the Answer in advance!

**Comment by reo7sp** [comment, marked as answer]
First enable inline mode for the bot in @BotFather. Then handle `InlineQuery`, create an article result, and pass it to `answerInlineQuery`:

```cpp
bot.getEvents().onInlineQuery([&bot](TgBot::InlineQuery::Ptr query) {
    auto text = std::make_shared<TgBot::InputTextMessageContent>();
    text->messageText = "You entered: " + query->query;

    auto content = std::make_shared<TgBot::InputMessageContent>();
    content->value = text;

    auto article = std::make_shared<TgBot::InlineQueryResultArticle>();
    article->id = "text-result";
    article->title = "Send result";
    article->inputMessageContent = content;

    auto result = std::make_shared<TgBot::InlineQueryResult>();
    result->value = article;

    bot.getApi().answerInlineQuery(query->id, {result}, 0, true);
});
```

The user invokes it as `@YourBot some text`. Telegram shows the article, and selecting it sends `You entered: some text`. There is no separate confirmation update after selection.
